### PR TITLE
[MRG]: read *.dat electrode position files

### PR DESCRIPTION
**Pull request by christianbrodbeck**
#### What does this implement/fix?
I implemented this in order to use electrode positions from the files provided [here](https://compumedicsneuroscan.com/scan-acquire-configuration-files/). I did not find an official format specification, so this is just based on looking at different files. If this is the right way of adding to the API I will improve documentation and maybe add a note that the reader might not be perfect.

### CODE REVIEW

**Comment by larsoner** [comment]
This looks reasonable to me. Maybe worth adding a test file? Or if this is a standard montage for the system (not individualized from an actual measurement) we could add to `mne/channels/data/montage`.

Also should be added to `python_reference.rst` and `latest.inc`. I would also probably add a `.. warning:: ` about the format being inferred from the file.

File: mne/channels/montage.py [hunk 1]
```diff
@@ -362,6 +362,49 @@ def transform_to_head(montage):
     return montage
 
 
+def read_dig_dat(fname):
+    r"""Read electrode positions from .dat file
```
Comment (larsoner) [inline mne/channels/montage.py]: Travis complains
```
mne/channels/montage.py:364 in public function `read_dig_dat`:
        D400: First line should end with a period (not 'e')
```
https://travis-ci.org/mne-tools/mne-python/jobs/593331023

**Comment by christianbrodbeck** [comment]
@larsoner As far as I understand these are standard systems (although I actually had to modify it slightly to work for the UMD linguistics EEG since it looks like they have some customization). I assume we can't just include the ones from the website though? I could email the company and ask for permission?

**Comment by agramfort** [comment]
@christianbrodbeck don't forget to update API page, add test, update the See Also from the other read_dig_xxx functions

**Comment by agramfort** [comment]
@christianbrodbeck can you also update the corresponding table in doc ? Grep to find where it is. Maybe in a tutorial

**Comment by christianbrodbeck** [comment]
in doc build, do you know if that is something l can fix?

‘’’
Makefile:49: recipe for target 'html_dev-pattern' failed
make: *** [html_dev-pattern] Error 1
Exited with code 2
‘’’

**Comment by drammock** [comment]
the problem is this
```
/home/circleci/project/mne/io/cnt/cnt.py:docstring of mne.io.read_raw_cnt:5: WARNING: py:meth reference target not found: mne.Raw.set_montage
``` (ctrl+f for "warning" because CircleCI elevates warnings to errors)

**Comment by christianbrodbeck** [comment]
Thanks @drammock !

**Comment by agramfort** [comment]
Thx @christianbrodbeck

**Pull request merged**